packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -35,6 +37,13 @@ const messageBaseVoid =
 const messageRejectionHandler =
   'A rejection handler that is not a function will be ignored.';
 
+const messagePromiseArray =
+  "An array of Promises may be unintentional. Consider handling the promises' fulfillment or rejection with Promise.all or similar.";
```

Comment:
Discussion is probably needed on the wording of these special case error messages. I just did my best.

Reply:
Yeah this is tricky. Thinking...

> of Promises

It might not be an array "of Promises". It might just happen to have one Promise.

> Promise.all

I hesitate to suggest this specific strategy because it's not always the right direction. Sometimes e.g. it's that they accidentally introduced async behavior in a loop that should be in series.

> Consider

We normally try to be a little more assertive in messages. Users get confused if we' seem not sure. 😄 

---

Maybe something like: _"This array may contain at least one floating Promise. Promises must be awaited, end with a call to .catch, or end with a call to .then with a rejection handler."_?

Reply:
Hmmmmm. My concern with this is that it doesn't give someone a way to solve it. Because if
```
[1, 2, Promise.resolve(), 3]
```
comes up, you can't solve it with
```
[1, 2, Promise.resolve().catch(() => {}), 3]
```

That's still a floating promise array (with the current impl at least). But you _could_ solve it with 
```
[1, 2, void Promise.resolve(), 3]; // effectively suppressed
Promise.all([1, 2, Promise.resolve(), 3]).catch(() => {}); // addressed
```
Basically the procedure is either
   a) don't put any promise in the array, _even if you have the handlers on it_
   b) wrap the  array with an array promise method, followed by resolving the ensuing unhandled promise

But I worry I would get misled into trying to resolve a) with the handlers, which is a red herring, and that I wouldn't know to do option b) at all.

Reply:
Yeah :/ I'm not a huge fan of the current wording for being a little too prescriptive too... maybe we can give a quick message and suggest reading the docs for more info?

Reply:
My idea for now...
```
This expression may produce an array containing a floating Promise. See the docs for more info on handling Promise arrays. 
```
Really struggling with coming up with something convincing haha.